### PR TITLE
From one file import all functions

### PR DESCRIPTION
**Pull request by littlewhitecloud**
Just support syntax like this:
```python
from ... import *
```

### CODE REVIEW

**Comment by littlewhitecloud** [comment]
just draft now, there is more things I need to do
- [x] Support syntax
- [ ] No error 
- [ ] Support import
- [ ] Check

**Comment by Akuli** [comment]
I'm not sure if I want this syntax. In languages that have it, it is usually considered bad style.

File: src/parse.c [hunk 1]
```diff
@@ -853,10 +853,15 @@ static void parse_import(const Token **tokens, const char *stdlib_path, Toplevel
...
+    if(parens) ++*tokens;
+    /*
+    if (!strcmp((*tokens)->data.name, '*') && (*tokens)->type != TOKEN_NAME)
+        printf("%d", !strcmp((*tokens)->data.name, '*'));
+        printf("\n");
+    */
 
     do {
-        if ((*tokens)->type != TOKEN_NAME)
+        if ((*tokens)->type != TOKEN_NAME && (*tokens)->type != TOKEN_OPERATOR) // Hah, failed many times make it to check *(tokens)->data.name == '*' successed
```
Comment (Akuli) [inline src/parse.c:864]: You can use the `is_operator()` function:
```suggestion
        if ((*tokens)->type != TOKEN_NAME && !is_operator(*tokens, "*"))
```

File: src/main.c [hunk 2]
```diff
@@ -320,9 +320,12 @@ static void add_imported_symbols(struct CompileState *compst)
         for (AstToplevelNode *ast = to->ast; ast->kind == AST_TOPLEVEL_IMPORT; ast++) {
             AstImport *imp = &ast->data.import;
             struct FileState *from = find_file(compst, imp->path);
-            assert(from);
+            assert(from);
 
-            for (struct ExportSymbol *es = from->pending_exports; es->name[0]; es++) {
+            for (struct ExportSymbol *es = from->pending_exports; es->name[0]; es++) {
+                if (es->name == '*') {
```
Comment (Akuli) [inline src/main.c:326]: Maybe:

```suggestion
                if (!strcmp(es->name, "*")) {
```

File: src/main.c [hunk 3]
```diff
@@ -320,9 +320,12 @@ static void add_imported_symbols(struct CompileState *compst)
...
             struct FileState *from = find_file(compst, imp->path);
-            assert(from);
+            assert(from);
 
-            for (struct ExportSymbol *es = from->pending_exports; es->name[0]; es++) {
+            for (struct ExportSymbol *es = from->pending_exports; es->name[0]; es++) {
+                if (es->name == '*') {
+                    printf("WOW, it successed!                                             sd");
+                }
                 if (!strcmp(imp->symbolname, es->name)) {
```
Comment (Akuli) [inline src/main.c:329]: I think this is the line that should be changed. If we are importing a symbol named `*`, it means we should match all exported symbols.

```suggestion
                if (!strcmp(imp->symbolname, es->name) || !strcmp(imp->symbolname, "*")) {
```

**Comment by Akuli** [comment]
Thanks for working on this. I might merge it later if I decide it is a good idea.

**Pull request closed without merging**